src/matrixrtc/EncryptionManager.ts

```diff
@@ -118,6 +126,9 @@ export class EncryptionManager implements IEncryptionManager {
         // as they may still be using the same ones.
         this.encryptionKeys.set(getParticipantId(this.userId, this.deviceId), []);
         this.transport.off(KeyTransportEvents.ReceivedKeys, this.onNewKeyReceived);
+        if (this.transport instanceof RoomAndToDeviceTransport) {
```

Comment:
I think we could avoid touching encryption manager by having the RoomAndToDeviceKeyTransport to cache the current senderkey in order to replay it to the new child transport.
This will be transparent to Ecryption manager